pkg/pipeline/common_steps.go

```diff
@@ -69,29 +74,49 @@ func addTenantLabel(obj PipelineObject, data *ExecutionContext) ExecutionResult
 
 func updateObject(obj PipelineObject, data *ExecutionContext) ExecutionResult {
 
-	oldResourceVersion := obj.GetObjectMeta().GetResourceVersion()
-
 	rtObj, ok := obj.(runtime.Object)
 	if !ok {
-		return ExecutionResult{
-			Abort: true,
-			Err:   fmt.Errorf("object ist not a valid runtime.object: %v", obj.GetObjectMeta().GetName()),
+		return ExecutionResult{Err: errors.New("object is not a runtime object")}
+	}
+
+	if !specAndMetaEqual(obj, data.originalObject) {
+		err := data.Client.Update(context.TODO(), rtObj)
```

Comment:
Like this we'd overwrite the `.status` of `rtObj` again, no?

Reply:
The `.status` of the `rtObj` is being merged back with the `patch()` below.